File: _a_arm/python_old/hw15/armParamHW15.py
# Single link arm Parameter File
import sys
sys.path.append('..')  # add parent directory
import armParam as P
import numpy as np
import control as cnt
from control import TransferFunction as tf
import matplotlib.pyplot as plt


# Compute plant transfer functions
th_e = 0;
Plant = tf([2.0/P.m/P.ell**2],
           [1, 2.0*P.b/P.m/P.ell**2, -3.0*P.g*np.sin(th_e)/2/P.ell])


# Bode plot of the plant
plt.figure(3), cnt.bode_plot(Plant, dB=True)

# Closes plot windows when the user presses a button.
plt.pause(0.0001)  # not sure why this is needed for both figures to display
print('Press key to close')
plt.waitforbuttonpress()
plt.close()

File: _A_arm/unsupported_languages/matlab/hw15/armParamHW15.m
% single link arm parameter file
addpath ./.. % adds the parent directory to the path
armParam % general parameters


% transfer function for robot arm
th_e = 0;
G = tf([2/P.m/P.ell^2],[1, 2*P.b/P.m/P.ell^2, -3*P.g*sin(th_e)/2/P.ell]);
figure(2), clf, bode(G), grid on

